### DISCUSSION TITLE
Contribution: lyngsat link compilation and more

### DISCUSSION THREAD

**Discussion by Dum4G** [iptv-org/community#249] [Show and tell]
I managed to gather all tvchannel related links from Lyngsat and also make human readable variant that looks like so

> ae/Abu Dhabi Drama
ae/Abu Dhabi Sports 1
ae/Abu Dhabi Sports 2
ae/Abu Dhabi Sports 3
ae/Abu Dhabi Sports 4
ae/Abu Dhabi TV Europe
ae/Abu Dhabi TV
ae/Ajman TV
ae/Al Aan TV
...

You can now automate sorting by country and many more
have fun!

[LYNG1.zip](https://github.com/iptv-org/iptv/files/6445464/LYNG1.zip)

**Comment by Dum4G** [comment]
Now sorted by channel names too
[alphabet.txt](https://github.com/iptv-org/iptv/files/6445929/alphabet.txt)

**Comment by Dum4G** [comment]
This is also a source with some great potential for automation
https://en.m.wikipedia.org/wiki/Lists_of_television_channels

**Comment by heroku-miraheze** [comment]
@freearhey

**Comment by freearhey** [comment]
I personally don't know how this might help, but if you have your own solution you can suggest it through a pull request.

**Comment by BellezaEmporium** [comment]
Could be useful if combined with a script that verifies the name of the channel being submitted and check on the Lyngsat text file to see if channel corresponds. A bot could then answer to the OP, saying where from which country he found it.

**Comment by BellezaEmporium** [comment]
@Dum4G Is there any way you can make a JSON out of this ? I could make a program that would read this JSON file and show all the channels with their names.

**Reply by Dum4G** [reply]
I have bigger plans for this. However, if you have an idea how it supposed to look like I may give it a try

**Reply by BellezaEmporium** [reply]
Something like this
{
   "ae": [{
      "insert name",
      "insert another name"
   ]},
  "al" ....
}

**Reply by Dum4G** [reply]
Nah, I don't see an easy way to transform my txt this way. You can make an app that will form it tho

**Reply by BellezaEmporium** [reply]
![image](https://user-images.githubusercontent.com/30985701/121720770-c23b8680-cae3-11eb-9e0b-77f70e23c226.png)

Done and done.

**Reply by BellezaEmporium** [reply]
Source code is here : https://github.com/LaneSh4d0w/lyngsat-channel-reader

**Reply by Dum4G** [reply]
> Source code is here : https://github.com/LaneSh4d0w/lyngsat-channel-reader

I bet your next goal is to add search and compile a release for Windows 😏 Nicely done

**Reply by BellezaEmporium** [reply]
> 
> 
> > Source code is here : https://github.com/LaneSh4d0w/lyngsat-channel-reader
> 
> I bet your next goal is to add search and compile a release for Windows 😏 Nicely done

I totally can, indeed. This program is built for Windows :)

**Reply by BellezaEmporium** [reply]
Actually @Dum4G , I got a better idea...

I think i'll make another program that asks for the different channels of this repo, read the unsorted m3u file, and compares the name with your Lyngsat list. Once he found the country, he'll copy all the data linked to the channel and replace it to the correct country.
